From 3ee7c80064a5ee7af3f0d6c1178138e1a2cf9085 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Wed, 28 Oct 2020 17:24:07 +0300
Subject: [PATCH 32/76] media: v4l2-async: workaround for RCAR VIN

This is a workaround for RCAR VIN drivers that use single media
device registration (notification complete) for all 8 VINs, but
This must be separated to VINs[0-3] and VINs[4-7] since these are
fully independents.
Currently if we have all 8 VIN enabled then any of quad VINs group
affect another one, i.e. if first groupd fail then sencond group fails
too due to miss of notification complete.
Normally the RCAR VIN must be reworked to separate 8 VINs into 2 independent
groups during registration.

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/media/v4l2-core/v4l2-async.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/v4l2-core/v4l2-async.c b/drivers/media/v4l2-core/v4l2-async.c
index e3ab003..6334ab2 100644
--- a/drivers/media/v4l2-core/v4l2-async.c
+++ b/drivers/media/v4l2-core/v4l2-async.c
@@ -246,8 +246,10 @@ v4l2_async_notifier_can_complete(struct v4l2_async_notifier *notifier)
 {
 	struct v4l2_subdev *sd;
 
+#if 0
 	if (!list_empty(&notifier->waiting))
 		return false;
+#endif
 
 	list_for_each_entry(sd, &notifier->done, async_list) {
 		struct v4l2_async_notifier *subdev_notifier =
-- 
2.7.4

